express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const json = require('./data.json');
+
+app.use(express.json());
+
+app.get('/api/notes', function (req, res) {
+  const temp = [];
+  for (const x in json.notes) {
+    temp.push(json.notes[x]);
+  }
+  res.status(200).send(temp);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const id = req.params.id;
+  const num = Number(req.params.id);
+  if (Number.isInteger(num) && num > 0) {
+    if (!json.notes[id]) {
+      const error = { error: `Cannot find Id of ${req.params.id}` };
+      res.status(404).send(error);
+    } else {
+      res.status(200).send(json.notes[id]);
+    }
+  } else if (req.params.id <= 0) {
```

Comment:
This should just be an `else` here, you already checked if it is an integer and greater than 0 above in the `if` statement, so there is no need to check again if it is less than 0